framework/quota/test/org/apache/cloudstack/quota/QuotaManagerImplTest.java

```diff
@@ -0,0 +1,203 @@
+    }
+
+    @Test
+    public void testAggregatePendingQuotaRecordsForAccount() {
+        AccountVO accountVO = new AccountVO();
+        accountVO.setId(2L);
+        accountVO.setDomainId(1L);
+        accountVO.setType(Account.ACCOUNT_TYPE_NORMAL);
+
+        UsageVO usageVO = new UsageVO();
+        usageVO.setQuotaCalculated(0);
+        usageVO.setUsageType(UsageTypes.ALLOCATED_VM);
+        List<UsageVO> usageVOList = new ArrayList<UsageVO>();
+        usageVOList.add(usageVO);
+        Pair<List<? extends UsageVO>, Integer> usageRecords = new Pair<List<? extends UsageVO>, Integer>(usageVOList, usageVOList.size());
+
+        QuotaUsageVO quotaUsageVO = new QuotaUsageVO();
+        quotaUsageVO.setAccountId(2L);
+        Mockito.doReturn(quotaUsageVO).when(quotaManager).updateQuotaAllocatedVMUsage(Mockito.eq(usageVO), Mockito.any(BigDecimal.class));
+
+        assertTrue(quotaManager.aggregatePendingQuotaRecordsForAccount(accountVO, new Pair<List<? extends UsageVO>, Integer>(null, 0)).size() == 0);
+        assertTrue(quotaManager.aggregatePendingQuotaRecordsForAccount(accountVO, usageRecords).size() == 1);
+    }
+
+    @Test
+    public void testProcessQuotaBalanceForAccount() {
+        AccountVO accountVO = new AccountVO();
+        accountVO.setId(2L);
+        accountVO.setDomainId(1L);
+        accountVO.setType(Account.ACCOUNT_TYPE_NORMAL);
+
+        QuotaUsageVO quotaUsageVO = new QuotaUsageVO();
+        quotaUsageVO.setAccountId(2L);
+        quotaUsageVO.setStartDate(new Date());
+        quotaUsageVO.setEndDate(new Date());
+        List<QuotaUsageVO> quotaListForAccount = new ArrayList<>();
+        quotaListForAccount.add(quotaUsageVO);
+
+        quotaManager.processQuotaBalanceForAccount(accountVO, quotaListForAccount);
+        Mockito.verify(quotaAcc, Mockito.times(1)).persistQuotaAccount(Mockito.any(QuotaAccountVO.class));
+    }
+
+    @Test
+    public void testUpdateQuotaRecords() {
+        UsageVO usageVO = new UsageVO();
+        usageVO.setId(100L);
+        usageVO.setQuotaCalculated(0);
+        usageVO.setUsageType(UsageTypes.NETWORK_BYTES_SENT);
+        usageVO.setRawUsage(9000000000.0);
+        usageVO.setSize(1010101010L);
+
+        QuotaTariffVO tariffVO = new QuotaTariffVO();
+        tariffVO.setCurrencyValue(new BigDecimal(1));
+        Mockito.when(quotaTariffDao.findTariffPlanByUsageType(Mockito.anyInt(), Mockito.any(Date.class))).thenReturn(tariffVO);
+
+        quotaManager.updateQuotaNetwork(usageVO, UsageTypes.NETWORK_BYTES_SENT);
+        quotaManager.updateQuotaAllocatedVMUsage(usageVO, new BigDecimal(0.5));
+        quotaManager.updateQuotaDiskUsage(usageVO, new BigDecimal(0.5), UsageTypes.VOLUME);
+        quotaManager.updateQuotaRaw(usageVO, new BigDecimal(0.5), UsageTypes.VPN_USERS);
+
+        Mockito.verify(quotaUsageDao, Mockito.times(4)).persistQuotaUsage(Mockito.any(QuotaUsageVO.class));
+        Mockito.verify(usageDao, Mockito.times(4)).persistUsage(Mockito.any(UsageVO.class));
```

Comment:
This test method contains no asserts.  How do we verify that the update of quota records succeeded rather simply not throwing an exception?